designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ScenarioActivityApiHttpService.scala

```diff
@@ -558,6 +576,15 @@ class ScenarioActivityApiHttpService(
     )
   }
 
+  private def markAttachmentAsDeleted(request: DeleteAttachmentRequest, scenarioId: ProcessId)(
+      implicit loggedUser: LoggedUser
+  ): EitherT[Future, ScenarioActivityError, Unit] =
+    EitherT(
+      dbioActionRunner.run(
+        scenarioActivityRepository.markAttachmentAsDeleted(scenarioId, request.attachmentId)
+      )
+    ).leftMap(_ => NoAttachment(request.attachmentId))
```

Comment:
all errors are mapped to NoAttachement? ScenarioNotFound too?

Reply:
Yes, that is actually the only error (I removed 2 other unused variants of DeleteAttachmentError). It is executed as a single db update, so we only get a count of updated rows.

Reply:
let's handle it explicitly. It'd be easier to catch changes in the future in the method declaration

Reply:
done